pkg/new-kernel/patches-5.15.x/0012-Add-support-for-LEDs-on-Operational-Services-EM320-d.patch

```diff
@@ -0,0 +1,244 @@
+From 154c5aa1acc0f47e489cfafa3caf5fcb3ff63bf0 Mon Sep 17 00:00:00 2001
+From: Mikhail Malyshev <mikem@zededa.com>
+Date: Wed, 31 Aug 2022 15:21:26 +0200
+Subject: [PATCH] Add support for LEDs on Operational Services EM320 device
+
+Signed-off-by: Mikhail Malyshev <mikem@zededa.com>
+---
+ drivers/platform/x86/Kconfig    |  10 ++
+ drivers/platform/x86/Makefile   |   4 +
+ drivers/platform/x86/os-em32x.c | 189 ++++++++++++++++++++++++++++++++
+ 3 files changed, 203 insertions(+)
+ create mode 100644 drivers/platform/x86/os-em32x.c
+
+diff --git a/drivers/platform/x86/Kconfig b/drivers/platform/x86/Kconfig
+index e21ea3d23e6f..3426071f87e7 100644
+--- a/drivers/platform/x86/Kconfig
++++ b/drivers/platform/x86/Kconfig
+@@ -1046,6 +1046,16 @@ config INTEL_SCU_IPC_UTIL
+ 	  low level access for debug work and updating the firmware. Say
+ 	  N unless you will be doing this on an Intel MID platform.
+ 
++config OS_EM32X
++	tristate "Operational Services  EM32x platform"
++	depends on X86_64
++	depends on I2C
++	depends on LEDS_CLASS
++	depends on LEDS_GPIO
++	depends on GPIO_PCA953X
++	help
++	  This driver add support for Operational Services EM32x platforms
++
+ endif # X86_PLATFORM_DEVICES
+ 
+ config PMC_ATOM
+diff --git a/drivers/platform/x86/Makefile b/drivers/platform/x86/Makefile
+index 2734a771d1f0..a95b911d87fd 100644
+--- a/drivers/platform/x86/Makefile
++++ b/drivers/platform/x86/Makefile
+@@ -119,3 +119,7 @@ obj-$(CONFIG_INTEL_SCU_PLATFORM)	+= intel_scu_pltdrv.o
+ obj-$(CONFIG_INTEL_SCU_WDT)		+= intel_scu_wdt.o
+ obj-$(CONFIG_INTEL_SCU_IPC_UTIL)	+= intel_scu_ipcutil.o
+ obj-$(CONFIG_PMC_ATOM)			+= pmc_atom.o
++
++# OS VW boxes
++obj-$(CONFIG_OS_EM32X)	+= os-em32x.o
++
+diff --git a/drivers/platform/x86/os-em32x.c b/drivers/platform/x86/os-em32x.c
+new file mode 100644
+index 000000000000..22fd703c712c
+--- /dev/null
++++ b/drivers/platform/x86/os-em32x.c
+@@ -0,0 +1,189 @@
++// SPDX-License-Identifier: GPL-2.0-or-later
++/*-*-linux-c-*-*/
++
++#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
++
++#include <linux/module.h>
++#include <linux/kernel.h>
++#include <linux/init.h>
++#include <linux/dmi.h>
++#include <linux/platform_device.h>
++#include <linux/i2c.h>
++#include <linux/platform_data/pca953x.h>
++
++#include <linux/leds.h>
++
++#define OS_DRIVER_VERSION "0.1"
++#define EM320_PCA4554_I2C_ADDR 0x22
++static struct pca953x_platform_data em320_led_gpio_data = {
++	.gpio_base = 0,
++	.invert = 0,
++	.irq_base = -1,
++};
++
++static struct i2c_board_info em320_board_info[] = { 
++	{
++		I2C_BOARD_INFO("pca9554", EM320_PCA4554_I2C_ADDR),
++		.platform_data = &em320_led_gpio_data,
++	} 
++};
++
++static struct gpio_led geos_leds[] = {
++	{
++		.name = "frontpanel:red:disk",
```

Comment:
Does something number these to get unique names in /sys/class/leds?

Reply:
I reviewed the latest Docimentation/leds/leds-class.rts one more time and now leds will be named as following:
red:disk-0 ... red:disk-3, blue:status-0 ... blue:status-3. The function name (disk, status) cannot be invented and since the led is not assosiated with other device e.g. NIC card the device name is ommited

Reply:
Do we know how they get numbered? Is it always predictable e.g., based on what pin on the UART they are connected to?

Reply:
I number them manually.